.circleci/config.base.yml

```diff
@@ -525,6 +528,36 @@ jobs:
           path: ~/repo/packages/amplify-migration-tests/amplify-migration-reports
     working_directory: ~/repo
 
+  amplify_migration_tests_v10:
+    parameters:
+      os:
+        type: executor
+        default: l_large
+    executor: << parameters.os >>
+    steps:
+      - restore_cache:
+          key: amplify-cli-repo-{{ .Branch }}-{{ .Revision }}
+      - restore_cache:
+          key: amplify-cli-yarn-deps-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - restore_cache:
+          key: amplify-pkg-binaries-{{ .Branch }}-{{ .Revision }}
+      - run:
+          name: Run tests migrating from CLI v10.5.1
+          command: |
+            source .circleci/local_publish_helpers.sh
+            changeNpmGlobalPath
+            cd packages/amplify-migration-tests
+            unset IS_AMPLIFY_CI
+            echo $IS_AMPLIFY_CI
```

Comment:
this is a small hack, unfortunately, in v10 there is custom logic that checks for this parameter, and will try to install the wrong version using a commit hash. That feature interferes with us trying to test a specific version, such as 10.5.1, but this was not an issue in v6/v5
In order for the script to download the exact version, we must disable this "commit hash" feature manually.
![image](https://user-images.githubusercontent.com/110861985/204679595-c858bf43-8147-4df5-bf5e-1eabaeabd5d0.png)